net/HttpRequest.hpp

```diff
@@ -831,7 +839,7 @@ class Response final
 
     StatusLine _statusLine;
     Header _header;
-    State _state; //< The state of the Response.
+    std::atomic<State> _state; //< The state of the Response.
```

Comment:
Is this accessed from multiple threads? I had the impression that most of your HTTP work is about having an async HTTP client which uses a single thread.

Reply:
It's true that async can still work on the same thread, but in practice the poll thread can be a different one, so it's best to make the state atomic. We typically handle the async result on the same poll thread, though.